common/middleware/local_node.py

```diff
@@ -0,0 +1,25 @@
+class LocalNodeMiddleware(object):
+    """
+    Ensures a Node that represents the local server always exists.
+
+    No other suitable hook for code that's run once and can access the server's host name
+    was found. A migration was not suitable for the second reason.
+    """
+    def __init__(self):
+        self.local_node_created = False
+
+    def process_request(self, request):
+        if not self.local_node_created:
+            from dashboard.models import Node
+            nodes = Node.objects.filter(local=True)
```

Comment:
This solution is acceptable within the context of this project but would see DB portability issues if sharing or migrating a server DB from one hostname address to another.